### PR TITLE
deps: Otel v0.55.0 Update

### PR DESCRIPTION
**Pull request by cpheps**
### Proposed Change
Updated Otel dependencies to v0.55.0

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Comment by BinaryFissionGames** [comment]
Looks like there were some deprecations that effect us in our OTel components (the processors, receivers, and exporters)
https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.55.0

All the component.WithX(receiver|exporter|processor) have been deprecated in favor of a function that also includes the stability level.

**Comment by cpheps** [comment]
>Looks like there were some deprecations that affect us in our OTel components (the resourceattributestransposerprocessor, pluginreceiver, and googlecloudexporter)
https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.55.0

>All the component.WithX(receiver|exporter|processor) have been deprecated in favor of a function that also includes the stability level.

Good catch I'll update.

File: receiver/pluginreceiver/emitter.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func createLogEmitterFactory(consumer consumer.Logs) component.ExporterFactory {
 	return component.NewExporterFactory(
 		emitterTypeStr,
 		defaultEmitterConfig,
-		component.WithLogsExporter(createExporter),
+		component.WithLogsExporterAndStabilityLevel(createExporter, component.StabilityLevelUndefined),
```
Comment (StefanKurek) [inline receiver/pluginreceiver/emitter.go:67]: so why are we sometimes saying beta and sometimes leaving it undefined?

File: receiver/pluginreceiver/go.mod [hunk 2]
```diff
@@ -42,18 +42,18 @@ require (
 	github.com/power-devops/perfstat v0.0.0-20210106213030-5aafc221ea8c // indirect
 	github.com/prometheus/client_golang v1.12.2 // indirect
 	github.com/prometheus/client_model v0.2.0 // indirect
-	github.com/prometheus/common v0.34.0 // indirect
+	github.com/prometheus/common v0.35.0 // indirect
 	github.com/prometheus/procfs v0.7.3 // indirect
 	github.com/prometheus/statsd_exporter v0.21.0 // indirect
 	github.com/shirou/gopsutil/v3 v3.22.6 // indirect
-	github.com/spf13/cobra v1.4.0 // indirect
+	github.com/spf13/cobra v1.5.0 // indirect
```
Comment (armstrmi) [inline receiver/pluginreceiver/go.mod:49]: Where do you find the documentation for updating this dependency and dependencies like it that don't seem to be listed in the changelog of the v0.55.0 release?

Comment (armstrmi) [reply receiver/pluginreceiver/go.mod:49]: So does it get changed upon `go mod tidy`?

Comment (cpheps) [reply receiver/pluginreceiver/go.mod:49]: Yeah, after updating the dependency `go mod tidy` takes care of this.

**Pull request merged**